sympy/physics/mechanics/kane.py

```diff
@@ -72,14 +76,18 @@ class KanesMethod(_Methods):
     configuration_constraints : iterable of Expr, optional
         Constraints on the system's configuration, i.e. holonomic constraints.
     u_dependent : iterable of dynamicsymbols, optional
-        Dependent generalized speeds.
+        Dependent generalized speeds. If nonlinear velocity constraints are used,
+        observe the comments further down.
     velocity_constraints : iterable of Expr, optional
         Constraints on the system's velocity, i.e. the combination of the
         nonholonomic constraints and the time-derivative of the holonomic
-        constraints.
+        constraints. They must be linear in q' and in u.
     acceleration_constraints : iterable of Expr, optional
         Constraints on the system's acceleration, by default these are the
         time-derivative of the velocity constraints.
+    nonlinear_velocity_constraints : iterable of Expr, optional
+        They may be nonlinear in u and q', but must be linear in the time
+        derivatives of u and of q'.
```

Comment:
```suggestion
        Constraints on the system's velocity. They may be nonlinear in u and
        q', but must be linear in the time derivatives of u and of q'.
```
Is it `may be nonlinear` or `should be nonlinear`?

Reply:
I changed as suggested.
I guess it would not hurt if they were linear - but of course itwould be silly to put them here if they are linear.
I was wondering about my sentence: _...but must be linear in the time derivatives of u and q'_ : no matter how nonlinear they are in u and q', they will be linerar in u' and q''. Should I remove this sentence? I guess it does not hurt.